src/blas/KokkosBlas_gesv.hpp

```diff
@@ -101,13 +104,27 @@ gesv (const AMatrix& A, const BXMV& B, const IPIVV& IPIV)
       "Valid options include zero-extent 1-D view (no pivoting), or 1-D View with size of "<< A0 << " (partial pivoting).";
     Kokkos::Impl::throw_runtime_exception (os.str ());
   }
-#ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS
+
+  // Check for no pivoting case. Only MAGMA supports no pivoting interface
+#ifdef KOKKOSKERNELS_ENABLE_TPL_MAGMA //have MAGMA TPL
+  #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS //and have BLAS TPL
+  if((!std::is_same< typename AMatrix::device_type::memory_space, Kokkos::CudaSpace >::value) &&
+     (IPIV0 == 0) && (IPIV.data()==nullptr)) {
+    std::ostringstream os;
+    os << "KokkosBlas::gesv: IPIV: " << IPIV0 << ". " <<
+      "BLAS TPL does not support no pivoting.";
+    Kokkos::Impl::throw_runtime_exception (os.str ());
+  }
+  #endif
+#else //not have MAGMA TPL
+  #ifdef KOKKOSKERNELS_ENABLE_TPL_BLAS //but have BLAS TPL
   if((IPIV0 == 0) && (IPIV.data()==nullptr)) {
     std::ostringstream os;
     os << "KokkosBlas::gesv: IPIV: " << IPIV0 << ". " <<
-      "TPL BLAS does not support no pivoting.";
+      "BLAS TPL does not support no pivoting.";
     Kokkos::Impl::throw_runtime_exception (os.str ());
   }
+  #endif
 #endif
```

Comment:
Can this be cleanup since only MAGMA supports no pivoting? We should be able to check for this runtime_error via a single ifdef and if via:
```c++
#if !defined(KOKKOSKERNELS_ENABLE_TPL_MAGMA)
  if(IPIV0 == 0) && (IPIV.data()==nullptr)
    // throw runtime_error
#endif
```

Reply:
@e10harvey Evan, I was thinking about that before but it looks like a single ifdef will not work. For example:
when ```KOKKOSKERNELS_ENABLE_TPL_MAGMA``` and ```KOKKOSKERNELS_ENABLE_TPL_BLAS``` are both ```ON``` and when the view is on host (i.e. want to use BLAS TPL interface), the no-pivoting case will not be checked if using a single ifdef. I think I should keep the current logic, although I don't like it much.